src/core.rs

```diff
@@ -0,0 +1,247 @@
+/// The main hash code used by Starknet.
+///
+/// Contains 251 bits of data and is generated by the pedersen hash function.
+#[derive(Clone, Debug, PartialEq)]
+pub struct StarkHash([u8; 32]);
```

Comment:
Is this `[u8; 32]` to match the `primitive_types::H256`?

Should this be `Copy`?

Reply:
Definitely `Copy`

Reply:
Not necessarily to match; but it does make it simpler for storage and the coming MPT stuff. If the `ff` derive accepted bytes I'd remove this type but `ff` seems to only want `u64`.

I'm always hesitant to mark types as `Copy` that aren't actually trivial to copy. As in, make the user use `clone()` to make the cost explicit. But maybe that's just a c++ PTSD issue, where copies and moves and copy-constructors make reasoning difficult.